File: src/app/services/contact.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Contact {
  id: number;
  fkIdLaboratoire: number;
  fkIdAdresse: number;
  numTel: string;
  fax: string;
  email: string;
  laboratoryName: string;  // Nom du laboratoire
  adresse: string;         // Adresse formatée
}

@Injectable({
  providedIn: 'root'
})
export class ContactService {

  private apiUrl = 'http://localhost:8055/contact';  // URL de l'API backend

  // Définir les en-têtes HTTP
  private headers = new HttpHeaders({
    'Content-Type': 'application/json',
  });

  constructor(private http: HttpClient) { }

  // Récupérer tous les contacts
  getAllContacts(): Observable<Contact[]> {
    return this.http.get<Contact[]>(`${this.apiUrl}/all`, { headers: this.headers });
  }

  // Récupérer un contact par son ID
  getContactById(id: number): Observable<Contact> {
    return this.http.get<Contact>(`${this.apiUrl}/${id}`, { headers: this.headers });
  }

  // Ajouter un nouveau contact
  addContact(contact: Contact): Observable<Contact> {
    return this.http.post<Contact>(`${this.apiUrl}/addContact`, contact, { headers: this.headers });
  }

  // Mettre à jour un contact
  updateContact(id: number, contact: Contact): Observable<Contact> {
    return this.http.put<Contact>(`${this.apiUrl}/updateContact/${id}`, contact, { headers: this.headers });
  }


  // Supprimer un contact
  deleteContact(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/deleteContact/${id}`, { headers: this.headers });
  }
  // Récupérer des contacts par ID de laboratoire
getContactsByLaboratoryId(fkIdLaboratoire: number): Observable<Contact[]> {
  return this.http.get<Contact[]>(`${this.apiUrl}/labo/${fkIdLaboratoire}`, { headers: this.headers });
}

}

File: contact-service/src/main/java/com/backend/contact_service/service/ContactService.java
package com.backend.contact_service.service;

import com.backend.contact_service.dto.ContactDTO;
import com.backend.contact_service.entity.Contact;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;


public interface ContactService {

    Contact addContact(ContactDTO contactDTO);

    Contact updateContact(Long id, ContactDTO contactDTO);

    List<Contact> getAllLContacts();

    Optional<Contact> getContactById(long id);

    void deleteContact(Long id);
<<<<<<< HEAD
    
    List<Contact> getContactsByLaboratoryId(Long fkIdLaboratoire);

=======

    public List<Contact> getContactsByLaboratoryId(Long fkIdLaboratoire);
>>>>>>> 2bec11be510752e94c7aee8b6585e9372c32062e
}
